src/miniscript/decode.rs

```diff
@@ -90,6 +90,20 @@ pub enum Terminal<Pk: MiniscriptKey, Ctx: ScriptContext> {
     // Elements
     /// `DEPTH <12> SUB PICK <num> EQUAL`
     Version(u32),
+    /// Prefix is initally encoded in the script pubkey
+    /// User provides a suffix such that hash of (prefix || suffix)
+    /// is equal to hashOutputs
+    /// Since, there is a policy restriction that initial pushes must be
+    /// only 80 bytes, we need user to provide suffix in separate items
+    /// There can be atmost 7 cats, because the script element must be less
+    /// than 520 bytes total in order to compute an hash256 on it.
+    /// Even if the witness does not require 7 pushes, the user should push
+    /// 7 elements with possibly empty values.
+    ///
+    /// CAT CAT CAT CAT CAT CAT <pref> SWAP CAT /*Now we hashoutputs on stack */
+    /// HASH256  
```

Comment:
trailing whitespace

Reply:
also lol @ this strategy. Sounds good to me.

Reply:
Actually, when parsing, we can actually count the number of `CAT`s and encode this in the terminal ... maybe in a followup PR we should do this. Because I think we usually only need one or two pushes rather than 7, and I think in all cases we know how many pushes we need.

Reply:
> and I think in all cases we know how many pushes we need.

We never know how many outputs the user would have in their transaction. So, it's best to keep it at 520 max?

Reply:
AH, I see. You mean that you num CATs should be 

`num_cats = (520 - prefix_len)//80`?

Reply:
Yeah